src/msg/async/ProtocolV2.cc

```diff
@@ -515,6 +515,7 @@ void ProtocolV2::stop() {
 
   std::lock_guard<std::mutex> l(connection->write_lock);
 
+  reset_recv_state();
```

Comment:
@liewegas I don't think this is correct because, in `connection->_stop()` a few lines below, it will call `dispatch_queue->discard_queue(conn_id)`, which will iterate through the messages still in queue (not yet dispatched) and will release the throttle bytes per each message.

Reply:
If you look at the v1 code it does the same thing...